src/chat.rs

```diff
@@ -187,10 +187,15 @@ impl<'a> Chat<'a> {
             .ok()
     }
 
-    pub unsafe fn get_profile_image(&self) -> Option<String> {
+    pub fn get_profile_image(&self) -> Option<String> {
```

Comment:
The rusty refactor here is to return an `Option<PathBuf>` that way you don't need to do the unsafe `dc_get_abs_path_safe(...).to_str().unwrap()`.  This unsafe conversion is then pushed to the ffi code where it is fine for this to live and you can use the `dc_tools::OsStrExt::to_c_string()` directly on the `PathBuf` result returned to handle these crazy conversions with less chance of blowing up (since converting a path to a c string can be done safer than converting a path to a rust string).

Reply:
done.

Reply:
ah, overlooked the second part of your comment ....